052-record-reconciled-versions.md

```diff
@@ -0,0 +1,186 @@
+  Once all other components installed, `Kafka` CR updates to:
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.37.0
+  ```
+- User can see `metadata.annotations[strimzi.io/reconciled] = 0.37.0` so they know it's reconciled to correct level
+
+
+### Upgrade (from no mechansim)
+Same as fresh install:
+
+- On an upgrade from 0.36.0, a user has a Kafka CR:
+  ```
+  strimzi.io/reconciled: null
+  strimzi.io/reconciling: null
+  ```
+- The Kafka Reconciler watcher picks up the CR and does an initial status update
+  ```
+  strimzi.io/reconciled: null
+  strimzi.io/reconciling: 0.37.0
+  ```
+  and creates `StrimziPodSet` CRs with annotations:
+  ```
+  strimzi.io/reconciled: null
+  strimzi.io/reconciling: null
+  ```
+- StrimziPodSet reconciler reconciles the SPS CR:
+  ```
+  strimzi.io/reconciled: null
+  strimzi.io/reconciling: 0.37.0
+  ```
+  and starts deploying Kafka brokers one at a time, and on completion updates the SPS CR to:
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.37.0
+  ```
+- Kafka Reconciler picks up that SPS is now Ready and finished, installing other components
+  Once all other components installed, Kafka CR updates to:
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.37.0
+  ```
+- User can see `metadata.annotations[strimzi.io/reconciled] = 0.37.0` so they know it's reconciled to correct level
+
+### Upgrade (with new mechansim)
+This defines an example where it is imagined that 0.37.0 had this mechanism already implemented, and is now upgrading to a new version 0.38.0
+
+- On an upgrade from 0.37.0, a user has a Kafka CR:
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.37.0
+  ```
+- The Kafka Reconciler watcher picks up the CR and does an initial status update
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.38.0
+  ```
+  and updates `StrimziPodSet` CR (not changing the annotations from previous, since Kafka reconciler does not own them) with:
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.37.0
+  ```
+- StrimziPodSet reconciler reconciles the SPS CR with annotations:
+  ```
+  strimzi.io/reconciled: 0.37.0
+  strimzi.io/reconciling: 0.38.0
+  ```
+  and starts deploying Kafka brokers one at a time, and on completion updates the SPS CR to:
+  ```
+  strimzi.io/reconciled: 0.38.0
+  strimzi.io/reconciling: 0.38.0
+  ```
+- Kafka Reconciler picks up that SPS is now `Ready` and finished, installing other components
+  Once all other components installed, Kafka CR updates to:
+  ```
+  strimzi.io/reconciled: 0.38.0
+  strimzi.io/reconciling: 0.38.0
+  ```
+- User can see `metadata.annotations[strimzi.io/reconciled] = 0.38.0` so they know the CR has been reconciled to correct level
+
+
+## Kafka Version Annotation
+As a follow-on from this proposal, the mechanism could be extended to cover Kafka versions as well, such as:
+```
+strimzi.io/kafkaVersion: 3.5.0
+strimzi.io/kafkaVersionUpdating: 3.4.0
```

Comment:
These versions would be the other way round wouldn't they? For moving from 3.4.0 to 3.5.0?